app/grandchallenge/workstations/templates/workstations/partials/session-control-data.html

```diff
@@ -0,0 +1 @@
+data-session-control {% for name, value in session_control_data %}{% if value is not False %} {{ name }}{% if value is not True %}="{{ value|stringformat:'s' }}"{% endif %}{% endif %}{% endfor %}
```

Comment:
`{% if value is not False %} {{ name }}{% if value is not True %}` makes it hard to follow. Did you copy this from elsewhere? I can't see where it would be used right now, but we could - does this mean that we could drop putting `data-session-control` in the template and adding `"data-session-control": True` to `data_attrs` in the context?

Reply:
Yes that is true, we could drop "data-session-control" from the template. I'm using the template that is used in Django forms for dropping in widget attributes. I just couldn't find a way to reference it directly, which is why I copied it over into a new partial template. The original template also takes the attributes from `widget.attrs`, which I changed here to just session_control_data for simplicity.

Reply:
`render_to_string('django/forms/attrs.html', ...)` ?

Reply:
I get a template does not exist error. I think we would need to extend the TEMPLATES["DIRS"] in settings.py, but I haven't found the correct reference yet.

Reply:
Ah, was looking at a jinja2 template. How about:

```python
render_to_string("django/forms/widgets/attrs.html", {"widget": "attrs": data_attrs})
```

Reply:
Still can't find the template :-/

Reply:
OK, this is interesting, I'm learning a lot about Django internals. See the document on the [Form Rendering API](https://docs.djangoproject.com/en/4.1/ref/forms/renderers/).

I think we could do:

```python
from django.forms.renderers import DjangoTemplates

...

DjangoTemplates().render("django/forms/attrs.html", {"attrs": data_attrs})
```

Alternatively I think we could add [`'django.forms'` to `INSTALLED_APPS`](https://docs.djangoproject.com/en/4.1/ref/forms/renderers/#django.forms.renderers.TemplatesSetting) but I'm unsure on the side effects of that.

Reply:
Yes, the DjangoTemplates renderer works!